IO/BeagleBone/CANBusBBB.cs

```diff
@@ -135,14 +180,53 @@ internal CANBusBBB(string CanName)
 		/// <returns>A tuple, with the first element being the ID of the received CAN frame and the second being the payload</returns>
 		public Tuple<uint, byte[]> Read()
 		{
-			CANFrame Frame = new CANFrame();
-			read(Socket, ref Frame, Marshal.SizeOf(Frame));
-			byte[] Payload = new byte[Frame.DataLength];
-			unsafe
+			if (Extended)
 			{
-				for (int i = 0; i < Frame.DataLength; i++) { Payload[i] = Frame.Data[i]; }
+				ExtendedCANFrame Frame = new ExtendedCANFrame();
+				read(Socket, ref Frame, Marshal.SizeOf(Frame));
+				byte[] Payload = new byte[Frame.DataLength];
+				unsafe
+				{
+					for (int i = 0; i < Frame.DataLength; i++) { Payload[i] = Frame.Data[i]; }
+				}
+				return new Tuple<uint, byte[]>(Frame.CANID, Payload);
+			}
+			else
+			{
+				CANFrame Frame = new CANFrame();
+				read(Socket, ref Frame, Marshal.SizeOf(Frame));
+				byte[] Payload = new byte[Frame.DataLength];
+				unsafe
+				{
+					for (int i = 0; i < Frame.DataLength; i++) { Payload[i] = Frame.Data[i]; }
+				}
+				return new Tuple<uint, byte[]>(Frame.CANID, Payload);
 			}
-			return new Tuple<uint, byte[]>(Frame.CANID, Payload);
+		}
+
+		private byte DLCToLen(byte CanDLC)
```

Comment:
Naming of this method. Is this a length? Then call it `Length`, not `Len`. Also comment as to what this does.